File: src/main/java/com/scienjus/smartqq/model/Friend.java
package com.scienjus.smartqq.model;

/**
 * 好友.
 *
 * @author ScienJus
 * @author <a href="http://88250.b3log.org">Liang Ding</a>
 * @date 2015/12/18.
 */
public class Friend {

    private long userId;

    private String markname = "";

    private String nickname;

    private boolean vip;

    private int vipLevel;

    @Override
    public String toString() {
        return "Friend{"
                + "userId=" + userId
                + ", markname='" + markname + '\''
                + ", nickname='" + nickname + '\''
                + ", vip=" + vip
                + ", vipLevel=" + vipLevel
                + '}';
    }

    public long getUserId() {
        return userId;
    }

    public void setUserId(long userId) {
        this.userId = userId;
    }

    public String getMarkname() {
        return markname;
    }

    public void setMarkname(String markname) {
        this.markname = markname;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public boolean isVip() {
        return vip;
    }

    public void setVip(boolean vip) {
        this.vip = vip;
    }

    public int getVipLevel() {
        return vipLevel;
    }

    public void setVipLevel(int vipLevel) {
        this.vipLevel = vipLevel;
    }

}

File: src/main/kotlin/com/scienjus/smartqqkotlin/model/Friend.kt
package com.scienjus.smartqqkotlin.model

import com.scienjus.smartqqkotlin.client.SmartQqClient
import java.util.*

/**
 * 好友.
 * @author ScienJus
 * @author [Liang Ding](http://88250.b3log.org)
 * @date 2015/12/18.
 */
data class Friend internal constructor(
        override val client: SmartQqClient,
        override val id: Long,
        internal val categoryIndex: Int,
        override val nickname: String?,
        val alias: String?,
        val status: String?,
        val clientType: Int?,
        val isVip: Boolean,
        val vipLevel: Int) : MessageTarget, User() {
    override val name: String?
        get() = nickname
    val category: FriendCategory
        get() = client.friendCategories[categoryIndex]

    override fun message(content: String) {
        client.message(SmartQqClient.TargetType.FRIEND, id, content)
    }

    private val info by lazy {
        client.getFriendInfo(id)
    }

    val bio: String?
        get() = info.bio

    val gender: String?
        get() = info.gender

    val phone: String?
        get() = info.phone

    val cellphone: String?
        get() = info.cellphone

    val email: String?
        get() = info.email

    val homepage: String?
        get() = info.homepage

    val birthday: Date
        get() = info.birthday

    val school: String?
        get() = info.school

    val job: String?
        get() = info.job

    val bloodType: Int
        get() = info.bloodType

    val country: String?
        get() = info.country

    val province: String?
        get() = info.province

    val city: String?
        get() = info.city

    val personal: String?
        get() = info.personal

    val shengxiao: Int
        get() = info.shengxiao

    val account: String?
        get() = info.account

    val vipInfo: Int
        get() = info.vipInfo
}